applications/StructuralMechanicsApplication/custom_advanced_constitutive/multi_linear_elastic_1d_law.cpp

```diff
@@ -0,0 +1,137 @@
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Klaus B. Sautter
+//
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/checks.h"
+#include "includes/properties.h"
+#include "custom_advanced_constitutive/multi_linear_elastic_1d_law.h"
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+//******************************CONSTRUCTOR*******************************************
+//************************************************************************************
+
+MultiLinearElastic1DLaw::MultiLinearElastic1DLaw()
+    : TrussConstitutiveLaw()
+{
+}
+
+//******************************COPY CONSTRUCTOR**************************************
+//************************************************************************************
+
+MultiLinearElastic1DLaw::MultiLinearElastic1DLaw(const MultiLinearElastic1DLaw& rOther)
+    : TrussConstitutiveLaw(rOther)
+{
+}
+
+//********************************CLONE***********************************************
+//************************************************************************************
+
+ConstitutiveLaw::Pointer MultiLinearElastic1DLaw::Clone() const
+{
+    return Kratos::make_shared<MultiLinearElastic1DLaw>(*this);
+}
+
+//*******************************DESTRUCTOR*******************************************
+//************************************************************************************
+
+MultiLinearElastic1DLaw::~MultiLinearElastic1DLaw()
+{
+    // TODO: Add if necessary
+}
+
+//*************************CONSTITUTIVE LAW GENERAL FEATURES *************************
+//************************************************************************************
+
+
+
+//************************************************************************************
+//************************************************************************************
+
+double& MultiLinearElastic1DLaw::CalculateValue(
+    ConstitutiveLaw::Parameters& rParameterValues,
+    const Variable<double>& rThisVariable,
+    double& rValue)
+{
+    if(rThisVariable == TANGENT_MODULUS){
+        Vector current_strain = ZeroVector(1);
+        rParameterValues.GetStrainVector(current_strain);
+        const double equivalent_strain = std::abs(current_strain[0]);
+        const Vector moduli_list(rParameterValues.GetMaterialProperties()[MULTI_LINEAR_ELASTICITY_MODULI]);
+
+        if (equivalent_strain>0.0){
+
+            const Vector strain_list(rParameterValues.GetMaterialProperties()[MULTI_LINEAR_ELASTICITY_STRAINS]);
+            const SizeType len_strain_list(strain_list.size());
+
+            SizeType counter(0);
+            for (SizeType i=0;i<len_strain_list;i++){
```

Comment:
++i is faster

Reply:
for POD types this does not matter, for iterators it does
hence good practice to use ++i all the time

Reply:
as this is true for iterators I don't see the benefits for an integer/SizeType

Reply:
alright